pwiz_tools/Skyline/Controls/Graphs/GraphSpectrum.cs

```diff
@@ -598,14 +603,44 @@ public bool ShouldShowMirrorPlot
 
         public IEnumerable<SpectrumDisplayInfo> AvailableSpectra
         {
-            get { return _spectra; }
+            get { return _spectra.SelectMany(s => s.Spectra); }
+        }
+
+        public class PrecursorSpectra
+        {
+            public PrecursorSpectra(TreeNodeMS selectedTreeNode, TransitionGroupDocNode precursor, SpectrumDisplayInfo[] spectra)
+            {
+                SelectedTreeNode = selectedTreeNode;
+                Precursor = precursor;
+                Spectra = spectra;
+            }
+
+            public void Add(IEnumerable<SpectrumDisplayInfo> spectra)
+            {
+                Spectra = (Spectra != null ? Spectra.Concat(spectra) : spectra).ToArray();
+            }
+
+            private TreeNodeMS SelectedTreeNode { get; }
+            public TransitionGroupDocNode Precursor { get; }
+            public SpectrumDisplayInfo[] Spectra { get; private set; }
```

Comment:
This should probably be a List<SpectrumDisplayInfo> because it's easier to add to a list than an array.